byron/semantics/executable-spec/src/Control/State/Transition/Trace/Generator/QuickCheck.hs

```diff
@@ -196,14 +197,27 @@ shrinkTrace
   => BaseEnv sts
   -> Trace sts
   -> [Trace sts]
-shrinkTrace baseEnv tr = interpretSTS @sts @traceGenEnv baseEnv
-    $ Trace.closure env st0 `traverse` traceSignalsShrinks
+shrinkTrace baseEnv tr =
+    interpretSTS @sts @traceGenEnv baseEnv
+    $ Trace.closure env st0 `traverse` shrunk
   where
     env = Trace._traceEnv tr
     st0 = Trace._traceInitState tr
-    signals = Trace.traceSignals Trace.OldestFirst tr
-    traceSignalsShrinks =
-      QuickCheck.shrinkList (shrinkSignal @sts @traceGenEnv) signals
+    signals = Trace.traceSignals Trace.NewestFirst tr
+
+    compareLen a b = compare (length a) (length b) -- ascending sort 
+    shrunk = sortBy compareLen (shrinkSignals signals)
+
+    shrinkSignals (sn:_last:[]) =
```

Comment:
This looks a bit convoluted. The problem is that the the shrinks are passed to `closure` in the reverse order. So isn't it sufficient to apply `reverse` to the these shrinks?

Reply:
here I'm not only fixing the order bug, but also writing a custom signal list shrinker (the goal of this PR). I've added commentary to the shrinkers, have a look and see if that clarifies my intent of my folly ;)